Source/WebCore/dom/WindowEventLoop.cpp

```diff
@@ -116,6 +116,11 @@ MicrotaskQueue& WindowEventLoop::microtaskQueue()
 }
 
 void WindowEventLoop::didReachTimeToRun()
+{
+    runTasks();
```

Comment:
Can we assert there that it's okay to execute scripts, and also that the stack is clean (i.e. we're not in the middle of executing EventLoop::run().

Reply:
We can add m_isRunning or something and use `SetForScope` to temporarily set that here.